### ISSUE TITLE
mypy fails to check abstractclassmethod

### ISSUE THREAD

**Issue by v-tronglh** [python/mypy#10861]
**Bug Report**

mypy raises `Missing positional argument` when checking `abstractclassmethod` but does not raise any errors when checking `classmethod`.

**To Reproduce**

1. I write a Python script named `test.py` as the following:
```python
from abc import ABC, abstractclassmethod
from typing import Any, Type


class Class:
    @classmethod
    def func(cls, n: Any) -> None:
        pass


class AbstractClass(ABC):
    @abstractclassmethod
    def func(cls, n: Any) -> None:
        pass


def call_func(class_obj: Type[Class], n: Any) -> None:
    class_obj.func(n)


def call_func_abstract(class_obj: Type[AbstractClass], n: Any) -> None:
    class_obj.func(n)
```
2. Then, I just run `mypy test.py`

**Actual Behavior**

Cheking ends up with error `Missing positional argument "n" in call to "func" of "AbstractClass"` in `call_func_abstract` but does not raise any errors in `call_func`.

**Environment**

- Mypy version used: mypy 0.910
- Mypy command-line flags: not used
- Mypy configuration options from `mypy.ini` (and other config files): not used
- Python version used: Python 3.8.10
- Operating system and version: Ubuntu MATE 20.04.2 LTS (Focal Fossa) 64-bit

**Comment by erictraut** [comment]
`abstractclassmethod` has been deprecated since Python 3.3. The recommended approach is to use decorator `@abstractmethod` with `@classmethod`. If you do this, it works fine in mypy.

**Comment by v-tronglh** [comment]
It works now, thanks for your help, @erictraut.
